cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1099,6 +1105,18 @@ def _add_head_node(self):
 
         return head_node_instance
 
+    def _get_head_node_role_name(self, head_node):
+        # Retrieve HeadNode role name
+        custom_instance_role_arns = head_node.iam.instance_role_arns
+        if custom_instance_role_arns:
+            # Case when custom instance profile or role is passed as input
+            # An instance profile can contain only one IAM role
```

Comment:
I would like to keep as it is so to be consistent with IAM API, since the [get-instance-profile](https://docs.aws.amazon.com/cli/latest/reference/iam/get-instance-profile.html) return a list of roles.

Reply:
refactored